### PR TITLE
Master builds stability label has been added

### PR DESCRIPTION
**Pull request by rzinurov**
Master builds stability label has been added

screen shot 2015-07-12 at 21 31 31

@rasmusbergpalm

### CODE REVIEW

File: src/jenkins/ui/MasterStatus.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ */
+export default class MasterStatus extends React.Component {
+
+    renderStability() {
+        var numSuccess = 0;
+        var numFailed = 0;
+        var masterBuilds = _.filter(this.props.builds, (build) => {
+            var onlyMaster = true;
+            _.each(BuildUtils.getRepoBranches(build.getParameters()), (branch) => {
+                if (branch && branch !== "master") {
```
Comment (rasmusbergpalm) [inline src/jenkins/ui/MasterStatus.js]: what about origin/master?

File: src/jenkins/ui/MasterStatus.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                    onlyMaster = false;
+                }
+            });
+            return onlyMaster;
+        });
+        _.forEach(masterBuilds, (build) => {
+            if (build.result === "SUCCESS") {
+                numSuccess++;
+            } else if (build.result === "FAILURE" || build.result === "UNSTABLE") {
```
Comment (rasmusbergpalm) [inline src/jenkins/ui/MasterStatus.js]: What about aborted builds?

Comment (rzinurov) [inline src/jenkins/ui/MasterStatus.js]: Aborted builds are ignored because I haven't found a better way to count them.

Comment (rzinurov) [inline src/jenkins/ui/MasterStatus.js]: Oh but I forgot about the time limit - does build get aborted in this case?

File: src/jenkins/ui/MasterStatus.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                numFailed++;
+            }
+        });
+        var stability;
+        if (numFailed === 0) {
```
Comment (rasmusbergpalm) [inline src/jenkins/ui/MasterStatus.js]: what if numsuccess+numfailed == 0
'100% (0/0)' seems a little weird

Comment (rzinurov) [inline src/jenkins/ui/MasterStatus.js]: What do you suggest in this case? `Unknown`?

File: src/jenkins/ui/MasterStatus.js (hunk 1 above)
Comment (rzinurov) [inline src/jenkins/ui/MasterStatus.js]: If you specify origin/master in the form it will fail badly.

Comment (rzinurov) [inline src/jenkins/ui/MasterStatus.js]: I've copied this code from build status view

File: src/jenkins/ui/MasterStatus.js (hunk 2 above)
Comment (rasmusbergpalm) [inline src/jenkins/ui/MasterStatus.js]: They should be ignored. 
There's a 90min timeout on the job itself
There's a 32min timeout on the subsets. If they time out it'll cause the main job to fail

File: src/jenkins/ui/MasterStatus.js (hunk 3 above)
Comment (rasmusbergpalm) [inline src/jenkins/ui/MasterStatus.js]: N/A

File: src/jenkins/ui/MasterStatus.js (hunk 1 above)
Comment (rasmusbergpalm) [inline src/jenkins/ui/MasterStatus.js]: Ok. And what about jobs that are started by using the normal integration-test-generic job? like this: https://bob.tradeshift.com/job/integration-test-generic-build/18410/

Comment (rasmusbergpalm) [inline src/jenkins/ui/MasterStatus.js]: If those work, then nevermind :)

**Comment by rasmusbergpalm** [comment]
If https://bob.tradeshift.com/job/integration-test-generic-build/18410/ is picked up correctly then :facepunch: and THANKS!

**Pull request merged**